lib/portlayer/storage/vsphere/volume.go

```diff
@@ -189,8 +203,26 @@ func (v *VolumeStore) VolumeCreate(ctx context.Context, ID string, store *url.UR
 	return vol, nil
 }
 
-func (v *VolumeStore) VolumeDestroy(ctx context.Context, ID string) error {
-	return fmt.Errorf("TBD.  Not supported yet")
+func (v *VolumeStore) VolumeDestroy(ctx context.Context, vol *storage.Volume) error {
+	if err := volumesInUse(ctx, vol.ID); err != nil {
+		log.Errorf("VolumeStore: delete error: %s", err.Error())
+		return err
+	}
+
+	volDir := v.volDirPath(vol.ID)
+
+	dstore, err := v.getDatastore(vol.Store)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("VolumeStore: Deleting %s", volDir)
```

Comment:
we could perform the in use check immediately before deletion instead of on entry to the function. Tiny difference, but still moves time of check closer to time of action.